gsa/src/web/pages/schedules/__tests__/detailspage.js

```diff
@@ -155,179 +145,149 @@ describe('Schedule Detailspage tests', () => {
-    expect(element).toHaveTextContent('Once');
+    expect(baseElement).toHaveTextContent('Recurrence');
+    expect(baseElement).toHaveTextContent('Once');
 
-    expect(element).toHaveTextContent('Duration');
-    expect(element).toHaveTextContent('Entire Operation');
+    expect(baseElement).toHaveTextContent('Duration');
+    expect(baseElement).toHaveTextContent('Entire Operation');
   });
 
-  test('should render user tags tab', () => {
+  test('should render user tags tab', async () => {
     const gmp = {
-      schedule: {
-        get: getSchedule,
-      },
-      permissions: {
-        get: getEntities,
-      },
       settings: {manualUrl, reloadInterval},
       user: {
         currentSettings,
         renewSession,
       },
     };
 
+    const [mock, resultFunc] = createGetScheduleQueryMock('foo', mockSchedule);
+    const [permissionMock, permissionResult] = createGetPermissionsQueryMock({
+      filterString: 'resource_uuid=foo first=1 rows=-1',
+    });
+    const [
+      renewSessionMock,
+      renewSessionResult,
+    ] = createRenewSessionQueryMock();
     const {render, store} = rendererWith({
       capabilities: caps,
       gmp,
       router: true,
       store: true,
+      queryMocks: [mock, permissionMock, renewSessionMock],
     });
 
     store.dispatch(setTimezone('CET'));
-    store.dispatch(setUsername('admin'));
 
-    store.dispatch(entityLoadingActions.success('12345', schedule));
+    const {baseElement} = render(<Detailspage id="foo" />);
 
-    const {baseElement} = render(<Detailspage id="12345" />);
+    await wait();
+
+    expect(resultFunc).toHaveBeenCalled();
+    expect(permissionResult).toHaveBeenCalled();
 
     const tabs = screen.getAllByTestId('entities-tab-title');
 
     expect(tabs[0]).toHaveTextContent('User Tags');
     fireEvent.click(tabs[0]);
 
-    expect(baseElement).toHaveTextContent('No user tags available');
-  });
-
-  test('should render permissions tab', () => {
-    const gmp = {
-      schedule: {
-        get: getSchedule,
-      },
-      permissions: {
-        get: getEntities,
-      },
-      settings: {manualUrl, reloadInterval},
-      user: {
-        currentSettings,
-        renewSession,
-      },
-    };
-
-    const {render, store} = rendererWith({
-      capabilities: caps,
-      gmp,
-      router: true,
-      store: true,
-    });
-
-    store.dispatch(setTimezone('CET'));
-    store.dispatch(setUsername('admin'));
-
-    store.dispatch(entityLoadingActions.success('12345', schedule));
-
-    const {baseElement} = render(<Detailspage id="12345" />);
+    await wait();
 
-    const tabs = screen.getAllByTestId('entities-tab-title');
+    expect(renewSessionResult).toHaveBeenCalled();
 
-    expect(tabs[1]).toHaveTextContent('Permissions');
-    fireEvent.click(tabs[1]);
-
-    expect(baseElement).toHaveTextContent('No permissions available');
+    expect(baseElement).toHaveTextContent('No user tags available');
   });
-
```

Comment:
I think the "should render permissions tab" tab test case has gone missing.

Reply:
Oops. Added it back.